src/inmanta/ast/statements/define.py

```diff
@@ -720,6 +762,12 @@ def __init__(self, name: LocatableString, toname: LocatableString) -> None:
         DefinitionStatement.__init__(self)
         self.name = str(name)
         self.toname = str(toname)
+        if "-" in self.toname:
```

Comment:
currently the import are done this way:
```
def p_import(p: YaccProduction) -> None:
    """import : IMPORT ns_ref"""
    p[0] = DefineImport(str(p[2]), str(p[2]))
    attach_lnr(p, 1)


def p_import_1(p: YaccProduction) -> None:
    """import : IMPORT ns_ref AS ID"""
    p[0] = DefineImport(str(p[2]), p[4])
    attach_lnr(p, 1)

```
So I think only a check on toname is needed if I remove the second str() 
in` p[0] = DefineImport(str(p[2]), str(p[2]))`

Reply:
Ok I see, i'll make the change :)

Reply:
I must admit I'm a bit conflicted on this. Overall I'm still in favor of raising the error here. Mainly because the error you're referring to is just a consequence of implementation details of `inmanta.module`. It is meant as a value error for a specific method dealing with module requirements, not as a sane report to an end-user compiling a model. Since it just arises from decisions/assumptions made in the project loading, refactors in `inmanta.module` might at any point result in code paths where this error is no longer raised in the case we're talking about. I see the current error more as an indirectly caused internal error that just happens to have a more or less relevant error message. My opinion is we should not depend on internal errors to protect our interfaces.

But like I said, I'm not sure about this myself, and I'm willing to drop this if @arnaudsjs and @wouterdb agree that this is redundant.

Reply:
@wouterdb rerequesting your review for this change.

Reply:
For me it's fine to drop it if the error message is sufficiently clear and helpful to the end-user.

Reply:
1. I like defense in depth, so the fact that it is checked elsewhere is not an argument to not check it here as well
2. We have to adhere to what is documented: according to the docs, do we allow a `-` in a module name? If not, exception, else, warning.